src/engine/framework/System.cpp

```diff
@@ -355,6 +392,144 @@ static void CloseSingletonSocket()
 #endif
 }
 
+#if defined(DAEMON_USE_FLOAT_EXCEPTIONS_AVAILABLE) && defined(__APPLE__)
+static void FloatingPointExceptionSignalHandler(int /* sig */, siginfo_t *sip, void* /* scp */)
+{
+#if defined(DAEMON_ARCH_arm64)
+	if (sip->si_code == ILL_ILLTRP)
+	{
+		Sys::Error( "Illegal trap" );
+	}
+#endif
+
+	Sys::Error( "Floating point exception" );
+}
+#endif
+
+static void SetFloatingPointExceptions()
+{
+#if defined(DAEMON_USE_FLOAT_EXCEPTIONS_AVAILABLE)
+	#if defined(DAEMON_ARCH_amd64) || defined(DAEMON_ARCH_i686)
+		#if defined(__USE_GNU) || defined(__FreeBSD__) || defined(__APPLE__)
+			int exceptions = 0;
+		#elif defined(_WIN32)
+			unsigned int exceptions = 0;
+		#endif
+
+		#if defined(DAEMON_USE_ARCH_INTRINSICS_i686_sse)
+			int sse_exceptions = 0;
+		#endif
+	#elif defined(DAEMON_ARCH_arm64)
+		#if defined(__APPLE__)
+			int arm_exceptions = 0;
+		#endif
+	#endif
+
+	// Operations with NaN.
+	if (common_floatExceptions_invalid.Get())
+	{
+		#if defined(DAEMON_ARCH_amd64) || defined(DAEMON_ARCH_i686)
+			#if defined(__USE_GNU) || defined(__FreeBSD__) || defined(__APPLE__)
+				exceptions |= FE_INVALID;
+			#elif defined(_WIN32)
+				exceptions |= _EM_INVALID;
+			#endif
+
+			#if defined(DAEMON_USE_ARCH_INTRINSICS_i686_sse)
+				sse_exceptions |= _MM_MASK_INVALID;
```

Comment:
amd64 Mac did not work (though arm64 did). I didn't get any effect from the 3 fault injectors.